### PR TITLE
added osuosl apt repo and tests

### PR DESCRIPTION
**Pull request by daviddjh**

### CODE REVIEW

**Comment by ramereth** [comment]
[apt-osuosl cookbook](https://trello.com/c/huS6Lutr/1572-apt-osuosl-cookbook)

File: recipes/default.rb [hunk 1]
```diff
@@ -15,3 +15,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+
+apt_repository 'osuosl' do
+  uri 'http://ftp.osuosl.org/pub/osl/repos/apt/'
+  components ['main']
```
Comment (ramereth) [inline recipes/default.rb]:
```suggestion
  components %w(main)
```

File: recipes/default.rb [hunk 2]
```diff
@@ -15,3 +15,11 @@
...
+  key 'http://ftp.osuosl.org/pub/osl/repos/apt/repo.gpg'
+  action :add
```
Comment (ramereth) [inline recipes/default.rb]: This is the default action, please remove.

File: recipes/default.rb [hunk 3]
```diff
@@ -15,3 +15,11 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
+
+apt_repository 'osuosl' do
+  uri 'http://ftp.osuosl.org/pub/osl/repos/apt/'
```
Comment (ramereth) [inline recipes/default.rb]:
```suggestion
  uri 'https://ftp.osuosl.org/pub/osl/repos/apt/'
```

File: recipes/default.rb [hunk 4]
```diff
@@ -15,3 +15,11 @@
...
+  components ['main']
+  key 'http://ftp.osuosl.org/pub/osl/repos/apt/repo.gpg'
```
Comment (ramereth) [inline recipes/default.rb]:
```suggestion
  key 'https://ftp.osuosl.org/pub/osl/repos/apt/repo.gpg'
```

**Review by ramereth — CHANGES_REQUESTED** [review]
Please so add ChefSpec tests for the default recipe.

**Review by ramereth — CHANGES_REQUESTED** [review]
Please do a major bump by doing ``!bump major``. You will have to manually create a PR in chef-repo adding this to any environment that needs it (likely only the workstation env).

**Review by ramereth — APPROVED** [review]

**Comment by daviddjh** [comment]
!bump major

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a major-level version bump to v1.0.0.  Have a nice day!

**Pull request merged**